### PR TITLE
[Interop] Improve RPC Codebase + Start Beacon Chain With Mock ETH1 Values

### PR DESCRIPTION
**Pull request by rauljordan**
This is part of #3398 

---

# Description

**Write why you are making the changes in this pull request**

This PR allows us to start the beacon chain and validator clients by mocking the eth1 data votes using a flag `--interop-mock-eth1data-votes`, which fills any of its instances with default values as specified by the interop documentation recommendation [here](https://github.com/ethereum/eth2.0-pm/issues/62)

**Improvements**

This PR also removes all instances of defined mocks and big interfaces in the rpc package, using instead mocks from the respective packages that are imported.

### CODE REVIEW

File: beacon-chain/rpc/proposer_server.go [hunk 1]
```diff
@@ -65,11 +70,38 @@ func (ps *ProposerServer) RequestBlock(ctx context.Context, req *pb.BlockRequest
 		return nil, errors.Wrap(err, "could not get parent block signing root")
 	}
 
-	// Construct block body
-	// Pack ETH1 deposits which have not been included in the beacon chain
-	eth1Data, err := ps.eth1Data(ctx, req.Slot)
-	if err != nil {
-		return nil, errors.Wrap(err, "could not get ETH1 data")
+	var eth1Data *ethpb.Eth1Data
+	if ps.mockEth1Votes {
```
Comment (rauljordan) [inline beacon-chain/rpc/proposer_server.go]: Review this

Comment (nisdas) [reply beacon-chain/rpc/proposer_server.go]: Can you add a test for this whole logic ? this path isnt tested

File: beacon-chain/rpc/service.go [hunk 2]
```diff
@@ -182,15 +173,16 @@ func (s *Service) Start() {
...
 		operationService: s.operationService,
 		p2p:              s.p2p,
-		chainService:     s.chainService,
+		attReceiver:      s.chainService,
+		headRetriever:    s.chainService,
 		cache:            cache.NewAttestationCache(),
 	}
 	validatorServer := &ValidatorServer{
 		ctx:                s.ctx,
 		beaconDB:           s.beaconDB,
```
Comment (nisdas) [inline beacon-chain/rpc/service.go]: you need to also intialize headFetcher here , maybe change it to headRetriever to make it consistent with the other services

File: beacon-chain/powchain/testing/BUILD.bazel [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    ],
+    importpath = "github.com/prysmaticlabs/prysm/beacon-chain/powchain/testing",
+    visibility = ["//beacon-chain:__subpackages__"],
+    deps = [
+        "//proto/eth/v1alpha1:go_default_library",
+        "//shared/bytesutil:go_default_library",
+        "//shared/event:go_default_library",
+        "//shared/trieutil:go_default_library",
+        "@com_github_ethereum_go_ethereum//common:go_default_library",
+    ],
```
Comment (prestonvanloon) [inline beacon-chain/powchain/testing/BUILD.bazel]: Please add `testonly = True`

File: beacon-chain/sync/testing/BUILD.bazel [hunk 4]
```diff
@@ -0,0 +1,8 @@
+load("@io_bazel_rules_go//go:def.bzl", "go_library")
+
+go_library(
+    name = "go_default_library",
+    srcs = ["mock.go"],
+    importpath = "github.com/prysmaticlabs/prysm/beacon-chain/sync/testing",
+    visibility = ["//beacon-chain:__subpackages__"],
```
Comment (prestonvanloon) [inline beacon-chain/sync/testing/BUILD.bazel]: Please add `testonly = True,`

**Pull request merged**